### PR TITLE
Rs chirality

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #

Changes made in this Pull Request:
 - now reads R/S chirality from RDKit molecules

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by cbouy** [comment]
I'll have a proper look next week but overall looking good!
For the tests you could directly `Universe.from_smiles` ?

File: package/MDAnalysis/converters/RDKitParser.py [hunk 1]
```diff
@@ -72,6 +73,18 @@
...
+def _rdkit_chiral_to_RS(val):
+    # convert RDKit chirality tag to R/S/''
+    from rdkit import Chem
+
+    if val == Chem.ChiralType.CHI_TETRAHEDRAL_CW:
+        return 'R'
+    elif val == Chem.ChiralType.CHI_TETRAHEDRAL_CCW:
+        return 'S'
+    else:
+        return ''
```
Comment (cbouy) [inline package/MDAnalysis/converters/RDKitParser.py]: So it turns out that `Chem.ChiralType.CHI_TETRAHEDRAL_CW` and CCW are based on the atoms order, which are not guaranteed to match with the CIP order according to [this thread](https://www.mail-archive.com/rdkit-discuss@lists.sourceforge.net/msg09255.html).

However, the chiral tags are available as an atom property when reading a molecule made from an isomeric SMILES or if `Chem.AssignStereochemistryFrom3D` was called, so everything is not lost!

```suggestion
def _rdkit_atom_to_RS(atom):
    """Fetches RDKit chiral tags"""
    try:
        return atom.GetProp("_CIPCode")
    except KeyError:
        return ""
```

File: package/MDAnalysis/converters/RDKitParser.py [hunk 2]
```diff
@@ -124,6 +138,8 @@ class RDKitParser(TopologyReaderBase):
     | atom.GetMonomerInfo().GetName()             | names                   |
     | atom.GetProp('_TriposAtomName')             |                         |
     +---------------------------------------------+-------------------------+
+    | atom.GetChiralTag()                         | chiralities             |
```
Comment (cbouy) [inline package/MDAnalysis/converters/RDKitParser.py]:
```suggestion
    | atom.GetProp('_CIPCode')                    | chiralities             |
```

File: package/MDAnalysis/converters/RDKitParser.py [hunk 3]
```diff
@@ -204,6 +223,7 @@ def parse(self, **kwargs):
             elements.append(atom.GetSymbol())
             masses.append(atom.GetMass())
             aromatics.append(atom.GetIsAromatic())
+            chiralities.append(_rdkit_chiral_to_RS(atom.GetChiralTag()))
```
Comment (cbouy) [inline package/MDAnalysis/converters/RDKitParser.py]:
```suggestion
            chiralities.append(_rdkit_atom_to_RS(atom))
```

File: package/MDAnalysis/core/selection.py [hunk 4]
```diff
@@ -567,6 +567,29 @@ def _apply(self, group):
...
+
+        self.values = vals
+
+    @return_empty_on_apply
+    def _apply(self, group):
+        attr = getattr(group, self.field)
+
+        mask = (attr == self.values[0])
+        for v in self.values[1:]:
+            mask |= (attr == v)
```
Comment (cbouy) [inline package/MDAnalysis/core/selection.py]: Any reason not to use np.isin ?
```suggestion
        mask = np.isin(attr, self.values)
```

File: testsuite/MDAnalysisTests/core/test_atomselections.py [hunk 5]
```diff
@@ -1405,3 +1406,36 @@ def test_unique_selection_on_ordered_group(u_pdb_icodes, sel, sort, ix):
...
+
+    m = Chem.MolFromSmiles(smi)
+    u = mda.Universe(m)
+
+    assert hasattr(u.atoms, 'chiralities')
+
+    assert u.atoms[0].chirality == ''
+    assert u.atoms[1].chirality == chirality
+
+    assert_equal(u.atoms[:3].chiralities, np.array(['', chirality, ''], dtype='U1'))
```
Comment (cbouy) [inline testsuite/MDAnalysisTests/core/test_atomselections.py:1426]: Isn't this a bit redundant? Or you want a specific test for the dtype?

File: package/MDAnalysis/core/selection.py (hunk 4 above)
Comment (cbouy) [reply package/MDAnalysis/core/selection.py]: Oh wait I'm not sure if self.values will be a string like `'R S'` or a list like `['R', 'S']`. It needs to be a list to work with np.isin

**Review by cbouy — CHANGES_REQUESTED** [review]
Could you also add a "roundabout" test ?
You start with an isomeric smiles to create an rdkit molecule, gather the chiral tags with `_rdkit_atom_to_RS` in an array, then convert the molecule to a Universe, then convert the Universe to an RDKit molecule, gather the chiral tags again and compare them.

File: package/MDAnalysis/core/selection.py (hunk 4 above)
Comment (richardjgowers) [reply package/MDAnalysis/core/selection.py]: `self.values` will be a list, as we consume all tokens until the next keyword.

File: testsuite/MDAnalysisTests/core/test_atomselections.py (hunk 5 above)
Comment (richardjgowers) [reply testsuite/MDAnalysisTests/core/test_atomselections.py:1426]: Yeah getting an array for an AtomGroup is a ilttle different to single values from an Atom, worth checking both.

**Comment by richardjgowers** [comment]
thanks @cbouy !

**Review by cbouy — COMMENTED** [review]
I think it's just missing some docs in `AtomGroup.select_atoms` and then we should be good!

**Comment by richardjgowers** [comment]
Cool done thanks @cbouy , @MDAnalysis/coredevs probably good to go now!

**Comment by IAlibay** [comment]
I've restarted CI to see if the py3.9 MacOS failures were just random.

File: package/CHANGELOG [hunk 6]
```diff
@@ -26,6 +26,7 @@ Fixes
 Enhancements
   * Add option for custom compiler flags for C/C++ on build and remove march
     option from setup.cfg (Issue #3428, PR #3429).
+  * Added R/S Chirality support to RDKit parsed molecules
```
Comment (IAlibay) [inline package/CHANGELOG:29]: tsk tsk tsk - newest entry first

File: package/MDAnalysis/core/selection.py [hunk 7]
```diff
@@ -567,6 +567,27 @@ def _apply(self, group):
...
+        self.values = vals
+
+    @return_empty_on_apply
+    def _apply(self, group):
+        attr = getattr(group, self.field)
+
+        mask = np.isin(attr, self.values)
+
+        return group[mask]
+
```
Comment (IAlibay) [inline package/MDAnalysis/core/selection.py:590]:
```suggestion


```

File: package/MDAnalysis/core/topologyattrs.py [hunk 8]
```diff
@@ -1957,6 +1957,13 @@ def _gen_initial_values(na, nr, ns):
         return np.zeros(na, dtype=bool)
 
 
+class RSChirality(AtomAttr):
+    "R/S chirality"
```
Comment (IAlibay) [inline package/MDAnalysis/core/topologyattrs.py]: Weird - are neither of these covered by tests?

Comment (IAlibay) [reply package/MDAnalysis/core/topologyattrs.py]:
```suggestion
    """R/S chirality"""
```
Mostly to keep to the style.

**Review by IAlibay — CHANGES_REQUESTED** [review]
Couple of tiny things - main this is I can't work out why codecov is complaining about coverage for those two topologyattr sections. I'll restart CI, if it's still happening afterwards could you look into it @richardjgowers ?

File: package/MDAnalysis/core/topologyattrs.py (hunk 8 above)
Comment (richardjgowers) [reply package/MDAnalysis/core/topologyattrs.py]: L1961 is a docstring, so I think this is just a codecov snafu

**Review by IAlibay — APPROVED** [review]
Still not sure what's up with that coverage issue, but 🤷🏽‍♂️

**Pull request merged**